spec/02-integration/04-admin_api/22-debug_spec.lua

```diff
@@ -0,0 +1,395 @@
+          method = "GET",
+          path = "/debug/node/log-level",
+        })
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        message = "log level: emerg"
+        return json.message == message
+      end, 30)
+
+      -- decrease log level to debug on both CP 1 and CP 2
+      res = assert(helpers.admin_client(nil, 9113):send {
+        method = "PUT",
+        path = "/debug/cluster/control-planes-nodes/log-level/debug"
+      })
+      body = assert.res_status(200, res)
+      json = cjson.decode(body)
+      message = "log level changed"
+      assert(json.message == message)
+
+      -- make sure we changed to debug on CP 1
+      res = assert(helpers.admin_client(nil, 9113):send {
+        method = "GET",
+        path = "/debug/node/log-level",
+      })
+      body = assert.res_status(200, res)
+      json = cjson.decode(body)
+      message = "log level: debug"
+      assert(json.message == message)
+
+      -- Wait for CP 2 to check for cluster events
+      helpers.wait_until(function()
+        -- make sure we changed to debug on CP 2
+        res = assert(helpers.admin_client(nil, 9114):send {
+          method = "GET",
+          path = "/debug/node/log-level",
+        })
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        message = "log level: debug"
+        return json.message == message
+      end, 30)
+    end)
+
+    it("common cluster endpoint not accepted in hybrid mode", function()
+      local res = assert(helpers.admin_client(nil, 9113):send {
+        method = "PUT",
+        path = "/debug/cluster/log-level/notice"
+      })
+      assert.res_status(404, res)
+    end)
+
+    it("newly spawned workers can update their log levels", function()
+      local res = assert(helpers.admin_client():send {
+        method = "PUT",
+        path = "/debug/node/log-level/crit",
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      local message = "log level changed"
+      assert(json.message == message)
+
+      -- make sure we changed to crit
+      res = assert(helpers.admin_client():send {
+        method = "GET",
+        path = "/debug/node/log-level",
+      })
+      body = assert.res_status(200, res)
+      json = cjson.decode(body)
+      message = "log level: crit"
+      assert(json.message == message)
+
+      local prefix = helpers.test_conf.prefix
+      assert(helpers.reload_kong(strategy, "reload --prefix " .. prefix))
+
+      -- Wait for new workers to spawn
+      helpers.wait_until(function()
+        -- make sure new workers' log level is crit
+        res = assert(helpers.admin_client():send {
+          method = "GET",
+          path = "/debug/node/log-level",
+        })
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+
+        message = "log level: crit"
+        return json.message == message
+      end, 30)
+    end)
+  end)
+end)
```

Comment:
we also need a test, probably just for a happy path only, that to ensure when we call
the API, the log is indeed printing message of given level. i know this is mostly
testing the underlying FFI is working, but we need to make sure it's e2e working
in kong.

Reply:
done: https://github.com/Kong/kong/blob/feat/FT-2818-dynamic_log_levels/spec/02-integration/04-admin_api/22-debug_spec.lua#L93-L141